app/frontend/src/components/UI/select/Select.tsx

```diff
@@ -42,6 +47,10 @@ const Select: React.FC<Props> = (props: Props) => {
   const [selectedOption, setSelectedOption] = useState<DropdownItem|undefined>();
   const ref = useRef(null);
 
+  useEffect(() => {
+    setSelectedOption(props.initialOption);
+  }, [ props.initialOption ]);
```

Comment:
It would run each time `initialOptions` is changed. Is it expected?

Reply:
`initialOption` is not supposed to change. It should only be set once